tools/src/main/java/org/apache/kafka/trogdor/coordinator/TaskManager.java

```diff
@@ -288,25 +291,83 @@ private JsonNode getCombinedStatus() {
-     * @param spec                  The specification of the task to create.
+     * @throws RequestConflictException - if a task with that ID and a different spec already exists
      */
-    public void createTask(final String id, TaskSpec spec)
-            throws Throwable {
+    public void createAndScheduleTasks(Map<String, TaskSpec> tasks) throws RequestConflictException, InvalidRequestException {
+        List<ManagedTask> managedTasks = new ArrayList<>();
+
+        Iterator<Map.Entry<String, TaskSpec>> it = tasks.entrySet().iterator();
+        while (it.hasNext()) {
+            Map.Entry<String, TaskSpec> idAndSpec = it.next();
+            String id = idAndSpec.getKey();
+            if (id.isEmpty())
+                throw new InvalidRequestException("Invalid empty ID in createTask request.");
+
+            TaskSpec spec = idAndSpec.getValue();
+            ManagedTask duplicateTask = this.tasks.get(id);
+            if (duplicateTask != null) {
+                if (!duplicateTask.spec.equals(spec))
+                    throw new RequestConflictException("Task ID " + id + " already " +
+                        "exists, and has a different spec " + duplicateTask.spec);
+                log.info("Task {} already exists with spec {}", id, spec);
+                if (tasks.size() == 1)
+                    return;
+                else
+                    it.remove();
+            }
+        }
+        for (Map.Entry<String, TaskSpec> idAndSpec: tasks.entrySet()) {
+            try {
+                ManagedTask createdTask = createTask(idAndSpec.getKey(), idAndSpec.getValue());
+                if (createdTask != null)
+                    managedTasks.add(createdTask);
+            } catch (Throwable e) {
+                log.info("Failed to create createTask(id={}, spec={}) error:", idAndSpec.getKey(), idAndSpec.getValue(), e);
+            }
+        }
+
+        boolean infoLogEnabled = managedTasks.size() < 5;
+        for (ManagedTask task: managedTasks) {
+            long delayMs = scheduleTask(task);
+            if (infoLogEnabled)
+                log.info("Created a new task {} with spec {}, scheduled to start {} ms from now.", task.id, task.spec, delayMs);
+            else
+                log.debug("Created a new task {} with spec {}, scheduled to start {} ms from now.", task.id, task.spec, delayMs);
+        }
+
+        log.info("Created {} tasks.", managedTasks.size());
+    }
+
+    private long scheduleTask(ManagedTask task) {
+        tasks.put(task.id, task);
+        long delayMs = task.startDelayMs(time.milliseconds());
+        task.startFuture = scheduler.schedule(executor, new RunTask(task), delayMs);
+        return delayMs;
+    }
+
+    /**
+     * @return a #{@link ManagedTask} instance or #{@code null} if there was an error creating the task
+     *  or if the task already exists
+     */
+    private ManagedTask createTask(String id, TaskSpec spec) throws Throwable {
```

Comment:
I don't think we really need this function any more... we can just submit to the executor from the other function.